packages/caravan-fees/src/cpfp.ts

```diff
@@ -258,44 +234,78 @@ export const createCPFPTransaction = (options: CPFPOptions): string => {
  * @returns {boolean} True if the combined fee rate meets or exceeds the target fee rate, false otherwise.
  *
  * @example
- * const parentFee = new BigNumber(1000);
- * const parentVsize = 200;
- * const childFee = new BigNumber(2000);
- * const childVsize = 150;
- * const targetFeeRate = 5;
- * const isSatisfied = isCPFPFeeSatisfied(parentFee, parentVsize, childFee, childVsize, targetFeeRate);
+ * const txAnalyzer = new TransactionAnalyzer({...});
+ * const childTxTemplate = new BtcTransactionTemplate({...});
+ * const isSatisfied = isCPFPFeeSatisfied(txAnalyzer, childTxTemplate);
  * console.log(isSatisfied); // true or false
  *
  * @throws {Error} If any of the input parameters are negative.
  */
 export function isCPFPFeeSatisfied(
-  parentFee: BigNumber,
-  parentVsize: number,
-  childFee: BigNumber,
-  childVsize: number,
-  targetFeeRate: number,
+  txAnalyzer: TransactionAnalyzer,
+  childTxTemplate: BtcTransactionTemplate,
 ): boolean {
   // Input validation
-  if (
-    parentFee.isNegative() ||
-    parentVsize < 0 ||
-    childFee.isNegative() ||
-    childVsize < 0 ||
-    targetFeeRate < 0
-  ) {
-    throw new Error("All input parameters must be non-negative.");
+  if (!txAnalyzer || !childTxTemplate) {
+    throw new Error("Invalid analyzer or child transaction template.");
   }
+  const parentFee = new BigNumber(txAnalyzer.fee);
+  const parentVsize = txAnalyzer.vsize;
+  const childFee = new BigNumber(childTxTemplate.currentFee);
+  const childVsize = childTxTemplate.estimatedVsize;
+  const targetFeeRate = parseFloat(txAnalyzer.cpfpFeeRate);
 
   const combinedFee = parentFee.plus(childFee);
   const combinedVsize = parentVsize + childVsize;
   const combinedFeeRate = combinedFee.dividedBy(combinedVsize);
+
   return combinedFeeRate.gte(targetFeeRate);
 }
+
+/**
+ * Validates that the combined fee rate of a parent and child transaction
+ * meets or exceeds the target fee rate for a Child-Pays-for-Parent (CPFP) transaction.
+ *
+ * This function calculates the combined fee rate of the parent transaction (from the analyzer)
+ * and its child transaction, then compares it to the target CPFP fee rate. It ensures
+ * that the CPFP transaction provides sufficient fee incentive for miners to include both
+ * transactions in a block.
+ *
+ * @param {TransactionAnalyzer} txAnalyzer - The analyzer containing parent transaction information and CPFP fee rate.
+ * @param {BtcTransactionTemplate} childTxTemplate - The child transaction template.
+ * @param {targetFeeRate}targetFeeRate - The Target fees set when we call the cpfp function.
+ * @param {boolean} strict - If true, throws an error when the fee rate is not satisfied. If false, only logs a warning.
+ * @returns {void}
+ *
+ * @throws {Error} If the combined fee rate is below the target fee rate in strict mode.
+ */
+export function validateCPFPPackage(
+  txAnalyzer: TransactionAnalyzer,
+  childTxTemplate: BtcTransactionTemplate,
+  targetFeeRate: number,
```

Comment:
Isn't the targetFeeRate already available from the txAnalyzer?

Reply:
Ahh ... yes sorry , It skipped my mind , correcting it , thanks for pointing that out

Reply:
correction [done](https://github.com/caravan-bitcoin/caravan/pull/114/commits/d0df3dc62b52c461ecd9a36b45baf4c0f5c1ddf5#diff-852fd83c241ed78594dc29389ec04d1425d784e665c706434a8de046f7daa85dR228)